Changed calculation of x position in grid view to avoid position jitter

martinhering opened the pull request:
I noticed some jitter when scaling items in grid layout. This is due to the x position being calculated accumulatively from the left edge of the view. I changed that to use the center of a column as the reference point for the x position.

jwilling commented:
Hi, thanks for your pull request! Sorry for missing this when it came through last week. :star2: 

I'm not exactly sure what's up with the diff. Is there any way you could rebase this based on the current master?

martinhering commented:
Actually I found an issue with my original positioning code for the grid layout. It's actually incompatible with the even padding option. In my fork, I marked itemPaddingEnabled as deprecated and removed support for that. I think for this type of behavior a flow layout would be the better solution. I also fixed the calculation of adjacent cells to account for vertical spacing and I completely reimplemented multiple selection with an option to turn it off. The problem was that it was not working with shift and option keys and the use of arrow keys was inconsistent with system behavior. So, I don't know if you also want these changes or not. I think only merging the original commit is not good for the reason above.
